Gordon360/Controllers/RecIM/ActivitiesController.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Services.RecIM;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    public class ActivitiesController : GordonControllerBase
+    {
+        private readonly IActivityService _activityService;
+
+        public ActivitiesController(IActivityService activityService)
+        {
+            _activityService = activityService;
+        }
+
+        ///<summary>Gets a list of all Activities</summary>
+        ///<param name="active"> Optional active parameter </param>
+        /// <returns>
+        /// All Existing Activities 
+        /// </returns>
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<Activity>> GetActivities(Boolean active = false)
+        {
+            if (active)
+            {
+                var result = _activityService.GetActivitiesByTime(null);
+                return Ok(result);
+            } else
+            {
+                var result = _activityService.GetActivities();
+                return Ok(result);
+            }
+        }
+
+        ///<summary>Gets a list of all Activities after given time</summary>
+        /// <returns>
+        /// All Existing Activities 
+        /// </returns>
+        [HttpGet]
+        [Route("date={time}")]
+        public ActionResult<IEnumerable<Activity>> GetAllActivities(DateTime time)
+        {
+            // Currently the parameter is a "DateTime" But it will be changed
+            // to a string, then parsed into a datetime object to be passed
+            // into service.
+            var result = _activityService.GetActivitiesByTime(time);
+            return Ok(result);
+        }
+
+        ///<summary>Gets a Activity object by ID number</summary>
+        /// <param name="activityID">League ID Number</param>
+        /// <returns>
+        /// Activity object
+        /// </returns>
+        [HttpGet]
+        [Route("{activityID}")]
+        public ActionResult<Activity> GetLeagueByID(int activityID)
+        {
+            var result = _activityService.GetActivityByID(activityID);
+            return Ok(result);
+        }
+
+
+        /// <summary>
+        /// Posts Activity into CCT.RecIM.Activity
+        /// </summary>
+        /// <param name="newActivity">League object with appropriate values</param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("")]
+        public async Task<ActionResult> CreateActivity(Activity newActivity)
+        {
+            await _activityService.PostActivity(newActivity);
+            return Ok();
```

Comment:
we could return an ID if we need it, but since the api data is fetched on render, im pretty sure you would be able to get the ID when needed (when you select the component)

Reply:
Firstly, it is generally best practice to return a 201 Created response with details of the new resource when it is created. Secondly, the only other way to get the ID would be to request the entire list of all activities again from the API - because in order to get a specific activity, you have to know the ID of that activity.

Reply:
Interesting, I'll return the ID in that case.